src/node_api.cc

```diff
@@ -191,15 +191,15 @@ class Reference : private Finalizer {
     delete reference;
   }
 
-  int Ref() {
+  uint32_t Ref() {
     if (++_refcount == 1) {
       _persistent.ClearWeak();
     }
 
     return _refcount;
   }
 
-  int Unref() {
+  uint32_t Unref() {
     if (--_refcount == 0) {
```

Comment:
Maybe I'm being paranoid, but I think this helper class should ensure there is not an underflow:
```C++
if (_refcount > 0) {
  --_refcount;
  ...
```